intake/cli/server/server.py

```diff
@@ -107,15 +112,33 @@ def get(self):
             else:
                 start = int(page_offset)
                 stop = int(page_offset) + int(page_size)
-            page = itertools.islice(cat.walk(depth=1).items(), start, stop)
+            if query_list:
+                # This could be a search of a search of a serach (etc.).
+                # Progressively apply each search and then page through the
+                # final results.
+                for args, kwargs in query_list:
+                    # The results of each search are cached.
+                    query_key += (str((args, kwargs)),)
+                    try:
+                        source_id = self.searches[query_key]
```

Comment:
Really, we want an LRU cache?

Reply:
Better yet, I think we can reuse the existing cache expiration mechanism and simplify this a bit. Done in 2bafa8ad1ecb7c1762226f2e923c150e4fe06596.